### PR TITLE
Add tox and Travis files for testing automation

### PR DESCRIPTION
**Pull request by jdufresne**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+script: tox
+
+matrix:
+  include:
+    - python: "3.6"
+      env: TOXENV=py36
+    - python: "3.7"
+      env: TOXENV=py37
+    - python: "3.6"
+      env: TOXENV=black
```
Comment (francoisfreitag) [inline .travis.yml]: Should linters run first to give feedback earlier when a PR does not match expected quality standards?

**Comment by jdufresne** [comment]
> Should linters run first to give feedback earlier when a PR does not match expected quality standards?

Definitely. That was my intention but flubbed the copy & paste. Fixed. Ready for review.

File: tox.ini [hunk 2]
```diff
@@ -0,0 +1,25 @@
+[tox]
+envlist =
+    black
+    flake8
+    isort
+    py{36,37}
+minversion = 1.9
```
Comment (francoisfreitag) [inline tox.ini]: Why specify a `minversion`?

Comment (jdufresne) [reply tox.ini]: Required for `skip_install` feature: https://tox.readthedocs.io/en/latest/config.html#conf-skip_install

**Pull request merged**